automation/run-tests.sh

```diff
@@ -146,6 +146,40 @@ function open_shell {
     run_exit
 }
 
+function is_file_changed {
+    git diff --exit-code --name-only origin/master -- $1
```

Comment:
IMHO it would be easier to understand the code if this function would return 0 if the file was changed.